tests/integration/targets/lookup_passwordstore/tasks/tests.yml

```diff
@@ -151,3 +168,86 @@
   assert:
     that:
       - readyamlpass == 'testpassword\nrandom additional line'
+
+- name: Create a password in a folder
+  set_fact:
+    newpass: "{{ lookup('community.general.passwordstore', 'folder/test-pass length=8 create=yes') }}"
+
+- name: Fetch password from folder
+  set_fact:
+    readpass: "{{ lookup('community.general.passwordstore', 'folder/test-pass') }}"
+
+- name: Verify password from folder
+  assert:
+    that:
+      - readpass == newpass
+
+- name: Try to read folder as passname
+  set_fact:
+    newpass: "{{ lookup('community.general.passwordstore', 'folder') }}"
+  ignore_errors: true
+  register: eval_error
+
+- name: Make sure reading folder as passname failed
+  assert:
+    that:
+      - eval_error is failed
+      - "\"passname folder not found\" in eval_error.msg"
+
+- name: Change passwordstore location explicitly
+  set_fact:
+    passwordstore: "{{ lookup('env','HOME') }}/.password-store"
+
+- name: Make sure password store still works with explicit location set
+  set_fact:
+    newpass: "{{ lookup('community.general.passwordstore', 'test-pass') }}"
+
+- name: Change passwordstore location to a non-existent place
+  set_fact:
+    passwordstore: "somenonexistentplace"
+
+- name: Try reading from non-existent passwordstore location
+  set_fact:
+    newpass: "{{ lookup('community.general.passwordstore', 'test-pass') }}"
+  ignore_errors: true
+  register: eval_error
+
+- name: Make sure reading from non-existent passwordstore location failed
+  assert:
+    that:
+      - eval_error is failed
+      - "\"Passwordstore directory 'somenonexistentplace' does not exist\" in eval_error.msg"
+
+- name: Move original pass out of the way
+  command:
+    argv:
+      - mv
+      - "{{ passpath }}"
+      - "{{ passpath }}.testorig"
+  args:
+    creates: "{{ passpath }}.testorig"
+
+- name: Create dummy pass script
+  ansible.builtin.copy:
+    content: |
+      #!/bin/sh
+      echo "ok"
+    dest: "{{ passpath }}"
+    mode: '0755'
+
+- name: Try reading from non-existent passwordstore location with different pass utility
+  set_fact:
+    newpass: "{{ lookup('community.general.passwordstore', 'test-pass') }}"
+
+- name: Try to read folder as passname with a different pass utility
+  set_fact:
+    newpass: "{{ lookup('community.general.passwordstore', 'folder') }}"
```

Comment:
Shouldn't you check the value of `newpass` (both here and above)?

Reply:
The test here is just to check the return code, so checking the value of `newpass` provides no real value - the point is that the call succeeds now where it failed before. It still might make sense to add it, so it follows the same pattern of the other checks and the reviewer/reader of the code won't have to wonder why it isn'tchecked ;)

Reply:
Alternatively add a comment, that will also help the next person trying to figure this out :)